### PR TITLE
botcom: inline rename

### PR DESCRIPTION
**Pull request by mimecuvalo**
Renaming is currently clunky with the dialog. This gets rid of the dialog and makes renaming inline.

https://github.com/user-attachments/assets/54dec16c-b0f8-467c-8547-b9d7392d1dcd

### Change type

- [ ] `bugfix`
- [ ] `improvement`
- [ ] `feature`
- [ ] `api`
- [x] `other`

### CODE REVIEW

File: apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopLeftPanel.tsx [hunk 1]
```diff
@@ -118,10 +132,18 @@ function TlaFileNameEditor({
 		(name: string) => {
 			setIsEditing(false)
 			onChange(name)
+			onEnd?.()
 		},
 		[onChange]
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopLeftPanel.tsx:137]:
```suggestion
		[onChange, onEnd]
```

File: apps/dotcom/client/src/tla/components/TlaFileMenu/TlaFileMenu.tsx [hunk 2]
```diff
@@ -19,16 +19,17 @@ import { copyTextToClipboard } from '../../utils/copy'
...
 export function TlaFileMenu({
 	children,
 	source,
 	fileId,
+	onRenameAction,
 }: {
 	children: ReactNode
 	source: string
 	fileId: TldrawAppFile['id']
+	onRenameAction: () => void
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/components/TlaFileMenu/TlaFileMenu.tsx:32]:
```suggestion
	onRenameAction(): void
```

File: apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx [hunk 3]
```diff
@@ -221,16 +230,91 @@ function TlaSidebarFileLink({ item }: { item: RecentFile }) {
...
 
 /* ---------------------- Menu ---------------------- */
 
-function TlaSidebarFileLinkMenu({ fileId }: { fileId: TldrawAppFile['id'] }) {
+function TlaSidebarFileLinkMenu({
+	fileId,
+	onRenameAction,
+}: {
+	fileId: TldrawAppFile['id']
+	onRenameAction: () => void
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx:314]:
```suggestion
	onRenameAction(): void
```

**Review by MitjaBezensek — APPROVED** [review]
Looks good, left a few smaller suggestions. 

What if we also started renaming on double click?

**Comment by mimecuvalo** [comment]
> What if we also started renaming on double click?

maaaaybe, i think some users are still used to dbl-clicking on things (:ahem: google drive :ahem:). so i'd be worried that a intent to navigate would accidentally lead to a rename 🤔

**Pull request merged**